tests/test_source.py

```diff
@@ -51,3 +51,19 @@ def test_esm_group(df, aggregation_dict):
     assert set(subset_df['member_id']) == set(ds['member_id'].values)
     source.close()
     assert source._ds is None
+
+
+def test_esm_group_empty_df(df, aggregation_dict):
+    empty_df = _get_subset(df)
```

Comment:
Not asking to change anything, but this line is confusing to me.  Maybe I haven't seen this type of test structure before.   I normally don't see a dataframe getting passed into a test, so I don't understand what value it has, hence why to expect it should be empty.

Reply:
Okay, I see how this may be confusing 

> I normally don't see a dataframe getting passed into a test, so I don't understand what value it has, hence why to expect it should be empty.

By default, `_get_subset()` returns an empty dataframe when the query is empty. So, calling `_get_subset(df)` is going to return an **empty** dataframe:
 
https://github.com/NCAR/intake-esm/blob/b906b7b3c7ba3e6fce6b0d15fa1b2cf1d5f9eec3/intake_esm/core.py#L623-L626

Reply:
I updated the test to use a more explicit option i.e. I manually created the empty dataframe.